src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1744,30 +1736,25 @@ else if (0 == requestedPacketSize)
                 }
             }
 
-            // Note booleanPropertyOn will throw exception if parsed value is not valid.
+            // Note isBooleanPropertyOn will throw exception if parsed value is not valid.
 
-            // have to check for null before calling booleanPropertyOn, because booleanPropertyOn
+            // have to check for null before calling isBooleanPropertyOn, because isBooleanPropertyOn
             // assumes that the null property defaults to false.
             sPropKey = SQLServerDriverBooleanProperty.SEND_STRING_PARAMETERS_AS_UNICODE.toString();
-            if (null == activeConnectionProperties.getProperty(sPropKey)) {
-                sendStringParametersAsUnicode = SQLServerDriverBooleanProperty.SEND_STRING_PARAMETERS_AS_UNICODE
-                        .getDefaultValue();
-            } else {
-                sendStringParametersAsUnicode = booleanPropertyOn(sPropKey,
-                        activeConnectionProperties.getProperty(sPropKey));
-            }
+            sendStringParametersAsUnicode = null == activeConnectionProperties.getProperty(
+                    sPropKey) ? SQLServerDriverBooleanProperty.SEND_STRING_PARAMETERS_AS_UNICODE.getDefaultValue()
+                              : isBooleanPropertyOn(sPropKey, activeConnectionProperties.getProperty(sPropKey));
 
             sPropKey = SQLServerDriverBooleanProperty.LAST_UPDATE_COUNT.toString();
-            lastUpdateCount = booleanPropertyOn(sPropKey, activeConnectionProperties.getProperty(sPropKey));
+            lastUpdateCount = isBooleanPropertyOn(sPropKey, activeConnectionProperties.getProperty(sPropKey));
             sPropKey = SQLServerDriverBooleanProperty.XOPEN_STATES.toString();
-            xopenStates = booleanPropertyOn(sPropKey, activeConnectionProperties.getProperty(sPropKey));
+            xopenStates = isBooleanPropertyOn(sPropKey, activeConnectionProperties.getProperty(sPropKey));
 
             sPropKey = SQLServerDriverStringProperty.RESPONSE_BUFFERING.toString();
-            responseBuffering = null;
-            if (activeConnectionProperties.getProperty(sPropKey) != null
-                    && activeConnectionProperties.getProperty(sPropKey).length() > 0) {
-                responseBuffering = activeConnectionProperties.getProperty(sPropKey);
-            }
+            responseBuffering = activeConnectionProperties.getProperty(sPropKey) != null
```

Comment:
This `if` statement was checking for empty string too, now it is only against `null`. 
`activeConnectionProperties.getProperty(sPropKey).length() > 0`

Reply:
good catch, thx